SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -383,7 +405,7 @@ - (SDLStreamHasBytesHandler)sdl_dataStreamHasBytesHandler {
         __strong typeof(weakSelf) strongSelf = weakSelf;
 
         uint8_t buf[[SDLGlobals globals].maxMTUSize];
-        while ([istream hasBytesAvailable]) {
+        while (istream.streamStatus == NSStreamStatusOpen && istream.hasBytesAvailable) {
```

Comment:
Yes, the same race condition can now occur with the data session because the dataStreamHasBytesHandler is executed on the IO thread and the accessory disconnect notification arrives on the main thread, causing data to be passed to the delegate while the main thread is tearing down the transport.